System/FilePath/Internal.hs

```diff
@@ -313,6 +313,15 @@ hasExtension :: FilePath -> Bool
 hasExtension = any isExtSeparator . takeFileName
 
 
+-- | Is the given string the final extension of the filename?
+--   The extension should not include the separator.
+--
+-- > "png" `isExtensionOf` "/directory/file.png"     == True
```

Comment:
@SwiftsNamesake yeah, it makes totally sense that you followed the existing convention; hence why I directed the question at @ndmitchell :-)